static/js/containers/pages/CheckoutPage_test.js

```diff
@@ -102,6 +104,134 @@ describe("CheckoutPage", () => {
+        {},
+        {
+          location: {
+            search: `product=${productId}`
+          }
+        }
+      )
+
+      sinon.assert.calledWith(
+        helper.handleRequestStub,
+        "/api/basket/",
+        "PATCH",
+        {
+          body:        { items: [{ id: productId }] },
+          credentials: undefined,
+          headers:     {
+            "X-CSRFTOKEN": null
+          }
+        }
+      )
+      assert.equal(inner.state().errors, hasError ? "error" : null)
+    })
+  })
+
+  describe("mount", () => {
+    const productError = "product error",
+      couponError = "coupon error",
+      couponCode = "codeforcoupon",
+      productId = 12345
+    let couponPayload, productPayload
+
+    beforeEach(() => {
+      couponPayload = {
+        body:        { coupons: [{ code: couponCode }] },
+        credentials: undefined,
+        headers:     {
+          "X-CSRFTOKEN": null
+        }
+      }
+      productPayload = {
+        body:        { items: [{ id: productId }] },
+        credentials: undefined,
+        headers:     {
+          "X-CSRFTOKEN": null
+        }
+      }
+    })
+    ;[
+      [true, false, couponError],
+      [true, true, null],
+      [false, false, productError],
+      [false, true, productError]
+    ].forEach(([hasValidProductId, hasValidCoupon, expError]) => {
+      it(`updates the basket with a ${
+        hasValidProductId ? "" : "in"
+      }valid product id and a ${
+        hasValidCoupon ? "" : "in"
+      }valid coupon code from the query parameter`, async () => {
+        if (!hasValidProductId) {
+          helper.handleRequestStub
+            .withArgs("/api/basket/", "PATCH", productPayload)
+            .returns({
+              status: 400,
+              body:   {
+                errors: "product error"
+              }
+            })
+        }
+        if (!hasValidCoupon) {
+          helper.handleRequestStub
+            .withArgs("/api/basket/", "PATCH", couponPayload)
+            .returns({
+              status: 400,
+              body:   {
+                errors: "coupon error"
+              }
+            })
+        }
+        const { inner } = await renderPage(
+          {},
+          {
+            location: {
+              search: `product=${productId}&code=${couponCode}`
+            }
+          }
+        )
+        // wait for componentDidMount to resolve
+        await Promise.resolve()
+        sinon.assert.calledWith(
+          helper.handleRequestStub,
+          "/api/basket/",
+          "PATCH",
+          productPayload
+        )
+        if (hasValidProductId) {
+          sinon.assert.calledWith(
+            helper.handleRequestStub,
+            "/api/basket/",
+            "PATCH",
+            couponPayload
+          )
+        } else {
+          sinon.assert.neverCalledWith(
+            helper.handleRequestStub,
+            "/api/basket/",
+            "PATCH",
+            couponPayload
+          )
+        }
```

Comment:
I'd prefer this `if/else` be changed to a single `assert` statement (`assert.equal(helper.handleRequestStub.calledWith(...), hasValidProductId)`) but you can take or leave that